File: 5.01.00.2163/src/utils_lgpl/deltares_common/tests/01_test/test_01.f90
!----- LGPL --------------------------------------------------------------------
!                                                                               
!  Copyright (C)  Stichting Deltares, 2011-2012.                                
!                                                                               
!  This library is free software; you can redistribute it and/or                
!  modify it under the terms of the GNU Lesser General Public                   
!  License as published by the Free Software Foundation version 2.1.                 
!                                                                               
!  This library is distributed in the hope that it will be useful,              
!  but WITHOUT ANY WARRANTY; without even the implied warranty of               
!  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU            
!  Lesser General Public License for more details.                              
!                                                                               
!  You should have received a copy of the GNU Lesser General Public             
!  License along with this library; if not, see <http://www.gnu.org/licenses/>. 
!                                                                               
!  contact: delft3d.support@deltares.nl                                         
!  Stichting Deltares                                                           
!  P.O. Box 177                                                                 
!  2600 MH Delft, The Netherlands                                               
!                                                                               
!  All indications and logos of, and references to, "Delft3D" and "Deltares"    
!  are registered trademarks of Stichting Deltares, and remain the property of  
!  Stichting Deltares. All rights reserved.                                     
!                                                                               
!-------------------------------------------------------------------------------
!  $Id: test_01.f90 1180 2012-01-13 17:05:48Z mourits $
!  $HeadURL: https://svn.oss.deltares.nl/repos/delft3d/tags/5.01.00.2163/src/utils_lgpl/deltares_common/tests/01_test/test_01.f90 $
subroutine test_01
!
!   Program to test some functions which are available in the util_mf libarray
!
!   util_mf: UTILities for Mor and Flow
!
!   Jan.Mooiman@deltares.nl
!   22 sep 05
!
    implicit none
    logical            :: error
    character(len=256) :: value
    character(len=20)  :: env (2)

    call util_getenv('D3D_HOME', value)
    write(*,*) 'Result D3D_HOME: <',trim(value),'>'

    env(1) = 'D3D_HOME'
    env(2) = 'd3d_home'
    value  = ' ' 
    call util_getenv(env(1), value)
    write(*,*) 'Result D3D_HOME: <',trim(value),'>'

!    call getmp(error, value)
!    write(*,*) 'Result getmp : ',error,'<',trim(value),'>'

end subroutine test_01

File: 5.01.00.2163/src/utils_lgpl/esmfsm/tests/test_01/test_01.c
//---- LGPL --------------------------------------------------------------------
//
// Copyright (C)  Stichting Deltares, 2011-2012.
//
// This library is free software; you can redistribute it and/or
// modify it under the terms of the GNU Lesser General Public
// License as published by the Free Software Foundation version 2.1.
//
// This library is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public
// License along with this library; if not, see <http://www.gnu.org/licenses/>.
//
// contact: delft3d.support@deltares.nl
// Stichting Deltares
// P.O. Box 177
// 2600 MH Delft, The Netherlands
//
// All indications and logos of, and references to, "Delft3D" and "Deltares"
// are registered trademarks of Stichting Deltares, and remain the property of
// Stichting Deltares. All rights reserved.
//
//------------------------------------------------------------------------------
// $Id: test_01.c 1180 2012-01-13 17:05:48Z mourits $
// $HeadURL: https://svn.oss.deltares.nl/repos/delft3d/tags/5.01.00.2163/src/utils_lgpl/esmfsm/tests/test_01/test_01.c $
#if defined(WIN32)
#  include <io.h>
#else
#  include <sys/types.h>
#  include <sys/wait.h>
#  include <unistd.h>
#endif

#include "esm.h"

#define DATASIZE        400
#define TRYS            10000
#define SLEEP           200000


void
JustDoIt (
    int     putter,
    int     esmContext
    ) {

    const char *  role = (putter == 0) ? "getter" : "putter";
    int     n;
    char    fn [100];
    FILE *  fh;
    char *  name = "data";
    void *  dataHandle;


    sprintf (fn, "results.%s", role);

    if (NULL == (fh = fopen (fn, "w"))) {
        printf ("%s cannot open output file\n", role);
        exit (1);
        }

    if (esm_connect (esmContext) != 0) {
        printf ("%s not connected to context %d\n", role, esmContext);
        exit (1);
        }

    if (putter) {
        dataHandle = esm_alloc (esmContext, name, DATASIZE);

        printf ("%s %screated at 0x%x\n", name, (dataHandle == NULL) ? "NOT " :
"", dataHandle);
        fflush (stdout);
        }

    else {      // getter
        int s = DATASIZE;
        /*int s = 0; */

        usleep (SLEEP);

        for (n = 0 ; n < TRYS ; n++) {
            printf ("Attempting access to %s\n", name);
            fflush (stdout);

            if (NULL != (dataHandle = esm_alloc (esmContext, name, s)))
                break;

            printf ("Wait #%d for %d usec\n", n+1, SLEEP);
            fflush (stdout);
            usleep (SLEEP);
            }

        if (n >= TRYS) {
            printf ("Too many waits...\n");
            goto end;
            }

        printf ("%s found at 0x%x\n", name, dataHandle);
        fflush (stdout);
        }

    end: {
        esm_prinfo (esmContext, fh);
        fclose (fh);
        return;
        }
    }


int
main (
    int     argc,
    char *  argv[]
    ) {

    int esmContext = -1;

    if (-1 == (esmContext = esm_create (4000000, 0))) {
        printf ("Cannot create context\n");
        exit (1);
        }

    printf ("Created context %d\n", esmContext);

    if (fork () == 0) {                 // child
        JustDoIt (0, esmContext);
        exit (0);
        }

    else {                              // parent
        JustDoIt (1, esmContext);
        wait (NULL);
        esm_delete (esmContext);
        printf ("Deleted context %d\n", esmContext);
        exit (0);
        }

    return 0;
    }
